include/ttmlir/Dialect/TTNN/IR/TTNNOpsAttrs.td

```diff
@@ -67,4 +67,14 @@ def TTNN_MemoryConfigAttr : TTNN_Attr<"MemoryConfig", "memory_config"> {
   let assemblyFormat = "`<` params `>`";
 }
 
+def TTNN_ShapeAttr : TTNN_Attr<"Shape", "shape"> {
+  let summary = "TTNN Shape attribute";
+  let description = [{
+    TTNN shape attribute
+  }];
+
+  let parameters = (ins ArrayRefParameter<"int64_t">:$shape);
+  let assemblyFormat = "`<` custom<DimensionList>($shape) `>`";
+}
```

Comment:
How is this different than the shape on the ranked tensor type?

Reply:
It represents a `ttnn::Shape` object. Sure, we could use the implicit shape, but I find this much easier to work with. It also allows us to create a "free-standing" shape object, which makes it easier to work with emitc and ttnn runtime.